### ISSUE TITLE
Documentation still implies RuleTester will log failures with zero valid or invalid tests

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#4681]
This can be reproduced in master easily by writing empty rule tests.

```js
var ruleTester = new RuleTester();
var rule = require("path/to/rule");

ruleTester.run("rule-name", rule, {
    valid: [
    ],

    invalid: [
    ]
});
```

Expected: Assertion error for lack of valid tests, assertion error for lack of invalid tests.
Actual: "0 passing"

**Comment by eslintbot** [comment]
@platinumazure Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I noticed this too while debugging some rules. It must be a regression, since it was working at some point.

**Comment by nzakas** [comment]
This was intentional. It used to error, but that meant you couldn't focus in on just one falling case to debug it, you'd always have to have one valid and one invalid. I don't see a great need for this type of check, especially as it makes it harder to debug.

**Comment by platinumazure** [comment]
Understood. Shall we remove references to that behavior in the docs and/or
code comments? I could file a PR for that under this issue if you like.

**Comment by nzakas** [comment]
Yeah, if there are still lingering references, we should definitely remove them.

**Comment by platinumazure** [comment]
Okay. I'll change the issue title when I'm at a PC.
